pkg/kv/iterators.go

```diff
@@ -317,8 +324,8 @@ func (p *PartitionIterator) Err() error {
 }
 
 func (p *PartitionIterator) Close() {
-	// check itr is set, can be null in case seek fails
-	if p.itr != nil {
+	// Check itr is set, can be null in case seek fails
+	if !p.itrClosed && p.itr != nil {
 		p.itr.Close()
 	}
```

Comment:
prefer to manage just one state - don't want to reference a option to nil
```suggestion
	// Check itr is set, can be null in case seek fails
	if !p.itrClosed {
		p.itr.Close()
		p.iterClosed = true
	}
```

Reply:
Done